README.md

```diff
@@ -10,59 +10,159 @@ A [JSON schema](api/schema.json) describing the report is also available.
+          "metadata": {},
           "suites": [],
           "attachments": [],
           "tests": [
             {
               "name": "yay",
               "status": "passed",
-              "startTime": "2021-11-09T21:10:59.550Z",
               "duration": 123,
-              "attachments": []
+              "startTime": "2023-06-19T18:26:06.227Z",
+              "attachments": [
+                {
+                  "name": "video.mp4",
+                  "path": "./video.mp4"
+                },
+                {
+                  "name": "screenshot2.png",
+                  "path": "./screenshot2.png"
+                }
+              ],
+              "metadata": {}
             },
             {
               "name": "nay",
               "status": "failed",
-              "startTime": "2021-11-09T21:10:59.550Z",
               "duration": 123,
-              "attachments": []
+              "output": "test failed",
+              "startTime": "2023-06-19T18:26:06.227Z",
+              "attachments": [
+                {
+                  "name": "video.mp4",
+                  "path": "./video.mp4"
+                }
+              ],
+              "metadata": {}
+            },
+            {
+              "name": "oops",
+              "status": "skipped",
+              "duration": 123,
+              "output": "test skipped",
+              "startTime": "2023-06-19T18:26:06.227Z",
+              "attachments": [],
+              "metadata": {}
             }
           ]
         }
       ],
-      "attachments": [],
+      "attachments": [
+        {
+          "name": "screenshot1.png",
+          "path": "./screenshot1.png"
+        }
+      ],
       "tests": []
     }
-  ]
+  ],
+  "metadata": {
+    "id": "123"
+  }
 }
 ```
+
+The resulting JUnit file of the above example is:
+```
+<testsuites status="failed" tests="3" failures="1" skipped="1" time="369">
+  <testsuite name="somegroup" status="failed" tests="0" failures="0" skipped="0" time="0">
+    <properties>
+      <property name="attachment" value="screenshot1.png">./screenshot1.png</property>
+    </properties>
+  </testsuite>
+  <testsuite name="somefile.test.js" status="failed" tests="3" failures="1" skipped="1" time="369">
+    <properties/>
+    <testcase name="yay" status="passed" time="123" timestamp="2023-06-19T18:26:06.227Z">
+      <properties>
+        <property name="attachment" value="video.mp4">./video.mp4</property>
+        <property name="attachment" value="screenshot2.png">./screenshot2.png</property>
+      </properties>
+    </testcase>
+    <testcase name="nay" status="failed" time="123" timestamp="2023-06-19T18:26:06.227Z">
+      <properties>
+        <property name="attachment" value="video.mp4">./video.mp4</property>
+      </properties>
+      <failure>
+        <![CDATA[test failed]]>
+      </failure>
+    </testcase>
+    <testcase name="oops" status="skipped" time="123" timestamp="2023-06-19T18:26:06.227Z">
+      <properties/>
```

Comment:
Hm, most test cases won't have any properties. It'd be better to avoid rendering out `<properties/>` in that case to avoid bloating the output, since this field is optional. So only render that field if there's at least one property.

Reply:
I just convert what metadata provided. If `<properties>` is not required here, then I think `metadata` field should be an optional one instead of a required field. I'll update `metadata`

Reply:
Will make it optional. While I'm just wondering the reason for the required `metadata` field. I thought this field is required so properties can be a required field since they are similar here.

Reply:
Good question! From the JS API perspective,`metadata` is not required. It's optional. That's why the constructor has a default value:
```
metadata: object = {}
```

More on that subject:
Most of the json fields in this reporter are defined to be empty, rather than undefined/nil values. This was done to increase the ease of handling the data further downstream, since you don't need the nil/undefined checks. The same logic _could_ be applied to the XML output. Though in that case, I'd rather lean towards what's more common/established in that ecosystem. We control the ecosystem for sauce-test-report.json and have more freedom there, but we don't in case of junit.

Reply:
That sounds reasonable. Thanks for the explanation!